### ISSUE TITLE
Use Go wiki rules

### ISSUE THREAD

**Issue by AlekSi** [golangci/misspell#11]
It would be nice if this linter reported, for example, "marshalling" (double ell) as incorrect spelling. See https://go.dev/wiki/Spelling

**Comment by ldez** [comment]
The following typos are already detected:
- cancelled
- marshalling
- canceling
- marshalled

Only the following typos are not detected:
- iff
- cancelation

To use US dictionary:
```bash
misspell -locale=US
```
or
```yml
linters-settings:
  misspell:
    locale: US
```

**Comment by ldez** [comment]
FYI I added a new option inside golangci-lint to add extra words https://github.com/golangci/golangci-lint/pull/4401

This feature will be available in the next release (v1.57.0) of `golangci-lint`

**Comment by ldez** [comment]
I added a new option `-dict`: the option takes a path to a CSV file.

<details>
<summary>config.example.csv</summary>

```csv
acounts,accounts
canidate,candidate
comfirm,confirm
commiter,committer
converage,coverage
currrently,currently
delimeter,delimiter
differrent,different
exclusing,excluding
finshed,finished
formated,formatted
inderect,indirect
insuficient,insufficient
likly,likely
mergable,mergeable
overrided,overridden
priortized,prioritized
registeration,registration
reuqest,request
reviwer,reviewer
superceded,superseded
underlaying,underlying
```

</details>

```console
$ misspell -dict config.example.csv ./temp
temp/temp.go:4:4: "acounts" is a misspelling of "accounts"
temp/temp.go:5:4: "canidate" is a misspelling of "candidate"
temp/temp.go:6:4: "comfirm" is a misspelling of "confirm"
temp/temp.go:7:4: "commiter" is a misspelling of "committer"
temp/temp.go:8:4: "converage" is a misspelling of "coverage"
temp/temp.go:9:4: "currrently" is a misspelling of "currently"
temp/temp.go:10:4: "delimeter" is a misspelling of "delimiter"
temp/temp.go:11:4: "differrent" is a misspelling of "different"
temp/temp.go:12:4: "exclusing" is a misspelling of "excluding"
temp/temp.go:13:4: "finshed" is a misspelling of "finished"
temp/temp.go:14:4: "formated" is a misspelling of "formatted"
temp/temp.go:15:4: "inderect" is a misspelling of "indirect"
temp/temp.go:16:4: "insuficient" is a misspelling of "insufficient"
temp/temp.go:17:4: "likly" is a misspelling of "likely"
temp/temp.go:18:4: "mergable" is a misspelling of "mergeable"
temp/temp.go:19:4: "overrided" is a misspelling of "overridden"
temp/temp.go:20:4: "priortized" is a misspelling of "prioritized"
temp/temp.go:21:4: "registeration" is a misspelling of "registration"
temp/temp.go:22:4: "reuqest" is a misspelling of "request"
temp/temp.go:23:4: "reviwer" is a misspelling of "reviewer"
temp/temp.go:24:4: "superceded" is a misspelling of "superseded"
temp/temp.go:25:4: "underlaying" is a misspelling of "underlying"
```

So I think I close this issue now.
